controllers/drplacementcontrol_controller.go

```diff
@@ -808,17 +808,20 @@ func (r *DRPlacementControlReconciler) Reconcile(ctx context.Context, req ctrl.R
 	return r.reconcileDRPCInstance(d, logger)
 }
 
-func (r *DRPlacementControlReconciler) setProgressionAndUpdate(
-	ctx context.Context, drpc *rmn.DRPlacementControl, progressionStatus rmn.ProgressionStatus,
+func (r *DRPlacementControlReconciler) setDeletionStatusAndUpdate(
+	ctx context.Context, drpc *rmn.DRPlacementControl,
 ) error {
-	updated := updateDRPCProgression(drpc, progressionStatus, r.Log)
+	updated := updateDRPCProgression(drpc, rmn.ProgressionDeleting, r.Log)
+	drpc.Status.Phase = rmn.Deleting
 
 	if updated {
 		if err := r.Status().Update(ctx, drpc); err != nil {
 			return fmt.Errorf("failed to update DRPC status: (%w)", err)
 		}
 	}
 
+	updateDRPCProgression(drpc, rmn.ProgressionDeleted, r.Log)
```

Comment:
update to delete/deleting happens only DRPC fails to delete and when there is an error which will get resolved in the next reconcile, so till then the progressions is marked as delete, so that user knows that the delete operation is in progress